app/controllers/repos_controller.rb

```diff
@@ -26,10 +26,16 @@ def show
 
       @charts = @benchmark_type.benchmark_result_types.map do |benchmark_result_type|
         cache_key = "#{BenchmarkRun.charts_cache_key(@benchmark_type, benchmark_result_type)}:#{@benchmark_run_display_count}"
+        cache_key_ruby_v = "#{cache_key}_ruby_v"
 
         if (columns = $redis.get(cache_key))
+          # read the cached @ruby_versions separately
+          @ruby_versions = JSON.parse($redis.get(cache_key_ruby_v))
```

Comment:
Good point, what about changing the if statement to `if (columns = $redis.get(cache_key)) && (ruby_versions = $redis.get(ruby_version_cache_key))`